src/engine/sparqlExpressions/QueryRewriteExpressionHelpers.h

```diff
@@ -37,6 +37,12 @@ struct GeoFunctionCall {
 std::optional<GeoFunctionCall> getGeoFunctionExpressionParameters(
     const SparqlExpression& expr);
 
+// Helper to check if `expr` is a `SparqlExpression` on the `geof:sf[Relation]`
+// function, given the templated `Relation`.
+template <SpatialJoinType Relation>
+std::optional<GeoFunctionCall> getGeoRelationExpressionParameters(
+    const SparqlExpression& expr);
+
```

Comment:
Actually, is this helper required to be declared in the header, or is it just an implementation detail that can be in an anonymous namespace in the corresponding `.cpp` file?